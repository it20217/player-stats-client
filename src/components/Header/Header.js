import { useContext } from "react";
import { Link, useNavigate } from "react-router-dom";
import AuthContext from '../../store/auth-context';
/** Styles */
import styles from "./Header.module.scss";
/** Images */
import Logo from "../../images/Logo.png";

function Header() { 

  const AuthCtx = useContext(AuthContext);
  const isLoggedIn = AuthCtx.isLoggedIn;

  let navigate = useNavigate();
  return(
    <div className={styles["Header"]}>
      <div 
        className={styles["img-container"]} 
        onClick={() => navigate('/')}
      >
        <img src={Logo} alt="Logo"/>
      </div>
      <div className={styles["text-logo"]}>
        <h1 className="text-2xl font-bold text-gray-900 pl-7">
          Football<span className="text-2xl font-normal italic"> academy</span> 
        </h1>
      </div>
      <div className={styles["btn-container"]}>
        {isLoggedIn 
          ?
            <button 
              onClick={() => {AuthCtx.logout(); navigate('/')}}
              className="mx-2 my-2 bg-white transition duration-150 ease-in-out hover:bg-slate-700 hover:border-slate-800 hover:text-yellow-400 rounded border border-slate-800 text-gray-800 px-10 py-4 text-sm hover:bg-slate-100 focus:outline-none focus:ring-2 focus:ring-offset-2  focus:ring-slate-100"
            >
              Logout
            </button>
          :
            <Link 
              to="/login"
              className="mx-2 my-2 bg-white transition duration-150 ease-in-out hover:bg-slate-700 hover:border-slate-800 hover:text-yellow-400 rounded border border-slate-800 text-gray-800 px-10 py-4 text-sm hover:bg-slate-100 focus:outline-none focus:ring-2 focus:ring-offset-2  focus:ring-slate-100"
            >
              Login
            </Link>
          }
      </div>
      
    </div>
  )
}

export default Header;