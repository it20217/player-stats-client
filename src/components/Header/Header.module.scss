
.Header {
  width: 100%;
  background-color: white;
  min-height: 8vh;

  display: flex;
  flex-direction: row;
  place-content: space-between;
}
.img-container {
  margin: auto;
  >img {
    margin: auto 20px;
    height: 6vmin;
  }
  :hover {
    cursor: pointer;
  }
}
.text-logo {
  width: 100%;
  text-align: left;
  align-self: center;
  margin-left: 16px;
}
.btn-container {
  margin: auto 20px ; 
}
