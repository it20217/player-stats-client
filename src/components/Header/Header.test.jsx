import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import Header from './Header';

describe("Header component",() => {
  it("renders correctly", () => {
    const component = render(
      <MemoryRouter initialEntries={["/"]}>
        <Header/>
      </MemoryRouter>
    );
    expect(component).toMatchSnapshot();
  });
});