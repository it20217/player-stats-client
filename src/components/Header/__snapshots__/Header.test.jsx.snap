// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Header"
      >
        <div
          class="img-container"
        >
          <img
            alt="Logo"
            src="Logo.png"
          />
        </div>
        <div
          class="text-logo"
        >
          <h1
            class="text-2xl font-bold text-gray-900 pl-7"
          >
            Football
            <span
              class="text-2xl font-normal italic"
            >
               academy
            </span>
          </h1>
        </div>
        <div
          class="btn-container"
        >
          <a
            class="mx-2 my-2 bg-white transition duration-150 ease-in-out hover:bg-slate-700 hover:border-slate-800 hover:text-yellow-400 rounded border border-slate-800 text-gray-800 px-10 py-4 text-sm hover:bg-slate-100 focus:outline-none focus:ring-2 focus:ring-offset-2  focus:ring-slate-100"
            href="/login"
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Header"
    >
      <div
        class="img-container"
      >
        <img
          alt="Logo"
          src="Logo.png"
        />
      </div>
      <div
        class="text-logo"
      >
        <h1
          class="text-2xl font-bold text-gray-900 pl-7"
        >
          Football
          <span
            class="text-2xl font-normal italic"
          >
             academy
          </span>
        </h1>
      </div>
      <div
        class="btn-container"
      >
        <a
          class="mx-2 my-2 bg-white transition duration-150 ease-in-out hover:bg-slate-700 hover:border-slate-800 hover:text-yellow-400 rounded border border-slate-800 text-gray-800 px-10 py-4 text-sm hover:bg-slate-100 focus:outline-none focus:ring-2 focus:ring-offset-2  focus:ring-slate-100"
          href="/login"
        >
          Login
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
